Software/app/src/main/java/org/foi/emp/collegecoursemanager/MainActivity.java

```diff
@@ -1,35 +1,74 @@
 package org.foi.emp.collegecoursemanager;
 
+import android.content.Context;
+import android.content.Intent;
 import android.os.Bundle;
 
 import com.google.android.material.floatingactionbutton.FloatingActionButton;
 import com.google.android.material.snackbar.Snackbar;
 
+import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.appcompat.widget.Toolbar;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProviders;
+import androidx.recyclerview.widget.ItemTouchHelper;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 
 import android.view.View;
-
 import android.view.Menu;
 import android.view.MenuItem;
 
+import org.foi.emp.collegecoursemanager.Adapters.KolegijAdapter;
+import org.foi.emp.collegecoursemanager.viewModels.BodoviViewModel;
+import org.foi.emp.collegecoursemanager.viewModels.KolegijViewModel;
+import org.foi.emp.core.Database.Database;
+import org.foi.emp.core.Entities.Kolegij;
+
+import java.util.List;
+
 public class MainActivity extends AppCompatActivity {
+    private Database database;
+    private RecyclerView recyclerView;
+    private KolegijViewModel kolegijViewModel;
+    private KolegijAdapter adapter;
+    private BodoviViewModel bodoviViewModel;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
         Toolbar toolbar = findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
-
+        this.kolegijViewModel = ViewModelProviders.of(this).get(KolegijViewModel.class);
+        this.database = Database.getInstance(this);
         FloatingActionButton fab = findViewById(R.id.btnDodajKolegij);
         fab.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
-                        .setAction("Action", null).show();
+                startActivity(new Intent(MainActivity.this, DodavanjeKolegija.class));
+            }
+        });
+        PostaviRecycleView();
+        makeKolegijEraseable(recyclerView,adapter);
+    }
+
+    private KolegijAdapter PostaviRecycleView() {
+        this.recyclerView = (RecyclerView) findViewById(R.id.recyleViewPopisKolegija);
+        this.adapter = new KolegijAdapter(this,this);
```

Comment:
Oko sokolovo, apply code formation.

```suggestion
        this.adapter = new KolegijAdapter(this, this);
```